WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Carriers and Rates/ShippingLabelCarriersAndRatesViewModel.swift

```diff
@@ -0,0 +1,82 @@
+import UIKit
+import SwiftUI
+import Yosemite
+
+final class ShippingLabelCarriersAndRatesViewModel: ObservableObject {
+
+    private let order: Order
+    private let originAddress: ShippingLabelAddress
+    private let destinationAddress: ShippingLabelAddress
+    private let packages: [ShippingLabelPackageSelected]
+    private let stores: StoresManager
+
+    enum SyncStatus {
+        case loading
+        case success
+        case error
+        case none
+    }
+
+    @Published private(set) var syncStatus: SyncStatus = .none
+
+    /// The rows observed by the main view `ShippingLabelCarriersAndRates`
+    ///
+    @Published private(set) var rows: [ShippingLabelCarrierRow] = []
+
+    var ghostRows: [ShippingLabelCarrierRow] {
+        var tempRows: [ShippingLabelCarrierRow] = []
+        for _ in 0..<3 {
+            let row = ShippingLabelCarrierRow(title: "Ghost title",
+                                              subtitle: "Ghost subtitle",
+                                              price: "Ghost price",
+                                              image: nil)
+            tempRows.append(row)
+        }
+        return tempRows
```

Comment:
ℹ️  This one also works. A bit shorter and no need for a temp variable. 

```suggestion
        Array(0..<3).map { _ in
            ShippingLabelCarrierRow(title: "Ghost title",
                                    subtitle: "Ghost subtitle",
                                    price: "Ghost price",
                                    image: nil)
        }

```

Reply:
Though I actually wonder if `ghostRows` even belongs in the VM since it's a UI implementation and not necessarily “data” that the VM is responsible for. In that sense, I also wonder if `ShippingLabelCarrierRow` itself should be returned by the `ViewModel` since that is a view (SwiftUI) and not data per se. I think I would have expected the VM to just _emit_ `ShippingLabelCarrierRate` models. And it's the view's responsibility to `map()` those models into `ShippingLabelCarrierRow`.

Just thinking out loud. No strong opinions at the moment. 😄

Reply:
If it sounds good to you, I can implement a super simple view model that will be pass to `ShippingLabelCarrierRow`, in this way the `ShippingLabelCarriersViewModel` will returns just the view model of the row, and the `ShippingLabelCarriers` will be responsible to map those models into `ShippingLabelCarrierRow`.

Reply:
That sounds better and seems to fit with how we used to do stuff in UIKit. Our `UITableViewCells` are always passed a `Model` of some sort. That simple `RowViewModel` could also have something like:

```swift
struct RowViewModel {
    /// If true, then use the fancy skeleton view
    let isGhost: Bool
}
```

Just a possibly stupid idea from me. 😄 

Totally fine to do this later when you have time. I'm not aware of the time constraints so don't wanna disrupt your flow. 🙂

Reply:
No problem at all, I'll do that right now ;)